app/jobs/scheduled_job.rb

```diff
@@ -1,34 +1,65 @@
 class ScheduledJob < ApplicationJob
+  class RetryableError < StandardError; end
+  class TooManyRetriesError < StandardError; end
+
   queue_as :default
 
   def perform
-    now = DateTime.now
-    beginning_of_hour = now.beginning_of_hour
-    end_of_hour       = now.end_of_hour
+    return if tasks.empty?
 
-    tasks = Scheduler.where(run_on: beginning_of_hour..end_of_hour)
+    tasks.each do |task|
+      run_task(task)
 
-    return unless tasks.size.positive?
+    rescue RetryableError
+      Rails.logger.warn("Task #{task.id} failed (attempt ##{task.attempts + 1}) with retryable error, rescheduling...")
+      task.reschedule!
 
-    tasks.each do |task|
-      vendor_module = "#{task.integration.vendor.camelize}Service::Batch".constantize
-      ctx = {
-        id: nil,
-        attributes: nil,
-        relationships: {
-          batch: {
-            data: {
-              id: task.batch_id
-            }
-          },
-          facility: {
-            data: {
-              id: task.facility_id
-            }
+    rescue TooManyRetriesError
+      report_error
```

Comment:
Should also trigger a `warning` email to the CS team each time the task is rescheduled.